### PR TITLE
fix(sdcm/utils/file.py): make read_lines_filtered iteration repitable

### PR DESCRIPTION
**Pull request by dkropachev**
To be able to re iterate result of read_lines_filtered again. 
For example when you read logs in cycle.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils/file.py [hunk 1]
```diff
@@ -39,6 +47,10 @@ def __init__(self, path: str, mode: str = 'r', buffering: Optional[int] = None,
     def get_file_length(self) -> int:
         return self._open().seek(0, 2)
 
+    def flush(self):
```
Comment (bentsi) [inline sdcm/utils/file.py:50]: where is it used?

File: sdcm/utils/file.py [hunk 2]
```diff
@@ -91,18 +103,20 @@ def readline(self, limit: int = -1) -> AnyStr:
...
-                if pattern.search(line):
-                    yield line
-                    break
+        def generator():
+            for line in self._io:
+                for pattern in patterns:
+                    if pattern.search(line):
+                        yield line
+                        break
+        return ReiterableGenerator(generator=generator)
```
Comment (bentsi) [inline sdcm/utils/file.py:119]: why do we need to reiterate?

File: sdcm/utils/file.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/utils/file.py:50]: I have used it in test

File: sdcm/utils/file.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/utils/file.py:119]: To be able to re iterate result of read_lines_filtered again. For example when you read logs in cycle.

Comment (bentsi) [reply sdcm/utils/file.py:119]: please check `itertools.cycle`

**Review by bentsi — APPROVED** [review]

**Pull request merged**